### ISSUE TITLE
Ellipsis not mentioned in docs

### ISSUE THREAD

**Issue by LukasDrude** [arogozhnikov/einops#22]
Great work!

I discovered in https://github.com/arogozhnikov/einops/blob/master/einops/einops.py#L199 that you also support ellipsis. Its an important feature so you may want to add it to the documentation.

**Comment by arogozhnikov** [comment]
Thanks!

I am not mentioning this in the docs for purpose as I expect this feature to be misused.
So, for now I'd better keep it as is.

**Comment by shoyer** [comment]
+1 I think support for ellipsis is noteworthy. In fact, I recently saw someone note support for ellipsis as a missing feature in einops!

It's true that it's probably a better idea to omit ellipsis when writing code specialized for particular shapes of arrays, but occasionally it's nice to write rank polymorphic code and there's no good alternative to it.

**Comment by zplizzi** [comment]
I also agree that this should be documented! Sometimes you have to reshape an array with an undefined number of axes, and I don't think there's any other way to do it with einops? Eg `rearrange(x, "a b ... -> (a b) ...")` seems to work as I would expect, and I'll be using this in my code right now to replace a messy `.reshape()`, so it's definitely not unusual.

**Comment by JeppeKlitgaard** [comment]
I was initially a little disappointed after not finding ellipsis support in the docs, as it was crucial for my application and I was really enjoying `einops` in other places. Very glad I found this issue – I hope the feature gets added to einops' excellent documention!

**Comment by lucasb-eyer** [comment]
`rearrange(x, "b ... -> b (...)")` is so much better than `np.reshape(x, (x.shape[0], np.prod(x.shape[1:])))` and it is a super widespread operation to do. Ellipsis is beautiful, consider mentioning it very late in the docs!

**Comment by NightMachinery** [comment]
I was just going to open a feature request for `...`. :D

I think the ellipsis notation can be turned even more powerful by adding the ability to name the axes group and add an explicit count. E.g.,
```
reduce(x, "b size[2] channel -> b size", "mean")
```

An undetermined count (what `...` currently does) can be shown using `[]` or `[-1]`.

**Comment by magehrig** [comment]
I agree with everyone. If there is a concern for misuse, we could have a paragraph informing the user about best practices.

**Comment by zjost** [comment]
I'm here because ellipsis are used in [StableDiffusion code](https://github.com/CompVis/stable-diffusion/blob/main/scripts/img2img.py#L234), and I had a hard time understanding since I couldn't find in the docs.

**Comment by Etienne-Meunier** [comment]
In the mean time you can find some usage of ellipsis in the einops paper : Rogozhnikov, Alex. "Einops: Clear and reliable tensor manipulations with einstein-like notation." International Conference on Learning Representations. 2021.

**Comment by yifeikong** [comment]
I'm here because ellipsis are used in FlashAttention. It's hard for non-native English speakers to know the word "ellipsis", let alone find the correct page to explain what it means in `einops`. I really hope this can be mentioned in the docs or tutorial.

**Comment by cleong110** [comment]
I would also like to point out that ellipsis/ three dots is used but not explained in the [official Pytorch examples](https://einops.rocks/pytorch-examples.html): 

[source code for the pytorch examples](https://github.com/arogozhnikov/einops/blob/main/docs/pytorch-examples.html#L450)

![image](https://github.com/user-attachments/assets/2877b23a-edb2-439d-8137-ed8614482b02)

**Comment by Conchylicultor** [comment]
I also use Ellipsis all the time. However, I'm not sure what's the reverse operation of:

```python
einops.rearrange(images, "... h w -> (...) h w")
```

How to specify `...` shape in the reverse operation ? Something like:

```python
einops.rearrange(images, "(...) h w -> ... h w", ellipsis=(batch, num_frame))
```

Is it supported ?

**Comment by arogozhnikov** [comment]
@Conchylicultor you want to use einops.pack/einops.unpack for this, not rearrange.

**Comment by Conchylicultor** [comment]
> @Conchylicultor you want to use einops.pack/einops.unpack for this, not rearrange.

I do not see how I can use `einops.pack`. It seems that `einops.pack`/unpack uses multiple inputs/outputs arrays.

Here I have a single array, that I want to reshape, so I think `einops.rearrange` is the right operation:

```python
images.shape == (b, n, h, w)
images  = einops.rearrange(images, "... h w -> (...) h w")
images.shape == (b*n, h, w)
```

How to specify `...` shape in the reverse operation ? Something like:

```python
images.shape == (b*n, h, w)
images = einops.rearrange(images, "(...) h w -> ... h w", ellipsis=(b, n))
images.shape == (b, n, h, w)
```
